jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -90,6 +92,40 @@ void validate_method_removal(
   REQUIRE(replacement_nondet_exists);
 }
 
+void validate_nondets_converted(
+  std::list<goto_programt::instructiont> instructions)
+{
+  bool nondet_exists = false;
+  bool allocate_exists = false;
+  for(const auto &inst : instructions)
+  {
+    // Check that our NONDET(<type>) exists on a rhs somewhere.
+    exprt target_expression =
+      (inst.is_assign()
+         ? to_code_assign(inst.code).rhs()
+         : inst.is_return() ? to_code_return(inst.code).return_value()
+                            : inst.code);
+
+    if(target_expression.id() == ID_side_effect)
+    {
+      auto side_effect = to_side_effect_expr(target_expression);
+      if(side_effect.get_statement() == ID_nondet)
+      {
+        nondet_exists = true;
+      }
+
+      if(side_effect.get_statement() == ID_allocate)
+      {
+        allocate_exists = true;
+      }
+    }
+  }
+
+  REQUIRE_FALSE(nondet_exists);
+  REQUIRE(allocate_exists);
+}
+#include<iostream>
```

Comment:
Debug includes? In any case should be at the top of the file (but probably deleted)